lib/mayaUsd/fileio/utils/readUtil.cpp

```diff
@@ -414,6 +417,14 @@ bool UsdMayaReadUtil::SetMayaAttr(
         MPlug& attrPlug,
         const VtValue& newValue)
 {
+    SdfValueTypeName plugValueType = Converter::getUsdTypeName(attrPlug,false);
+    const auto* converter = Converter::find(plugValueType, attrPlug.isArray());
+    if(converter && newValue.GetType() == plugValueType.GetType()) {
```

Comment:
How common a pattern do you think this typeName checking is going to be in client code?

It seems a bit cumbersome, so if it's going to be everywhere then I'm wondering whether it might make sense to push it down into the converters themselves, and maybe have `Converter::convert()` return a `bool` value instead of `void`, returning `true` for a successful conversion and `false` otherwise.

Reply:
There shouldn't be. IIRC, this is where we double to float conversion...and I wanted to filter it out for now. Clearly we need to decide how we will be dealing with double to float conversion.